src/renate/training/training.py

```diff
@@ -515,7 +515,7 @@ def _execute_training_and_tuning_job_locally(
     config_space["updater"] = updater
     config_space["max_epochs"] = max_epochs
     config_space["config_file"] = config_file
-    config_space["prepare_data"] = False
+    config_space["prepare_data"] = True
```

Comment:
nice find. this was probably introduced with my latest change. you could try `"False"` instead and maybe that works. Otherwise, I would need to dive deeper into this. It should not be True since otherwise we keep downloading data for each new training job.

Background: Previously this value was set to 0 and we used argparse to read it as an int. Now, this is considered an argsparse str and we should correctly convert it to a bool. Since this is not happening, it might be that 1) input is wrong or 2) that there might be a deeper problem that will cause problems with all bool inputs. My guess right now is option 3: something is going wrong pre-cli logic.

Reply:
ok, I think it's much simpler than I've thought. there is a problem with the nlp dataset class. It seems to assume that prepare_data is always called before setup. however, the assumption is that prepare_data is only called once on the machine. based on my understanding following happens: 1) dataset object is created and prepare_data is called 2) another dataset object is created, only setup is called and this one is used for training. setting prepare_data to True will obviously "fix" this but I think we had a different understanding about the interface.

Reply:
You are absolutely right. I'll fix this. Thanks for catching this.